Source/ACE/Network/NetworkSession.cs

```diff
@@ -168,24 +181,22 @@ private void FlushBundle()
 
         private void FlushPackets()
         {
-            while (PacketQueue.Count > 0)
+            while (packetQueue.Count > 0)

```

Comment:
needs to lock packetQueueLock and copy out the items to be sent and/or "new" up the queue from within the lock, then continue.